### PR TITLE
[gui] add container.actualitems infolabel

### PR DESCRIPTION
**Pull request by phil65**
Absolute amount of items in a container (including parentfolder item).
Much more helpful than Container.NumItems for several use cases.
@HitcherUK 	@ronie 	@BigNoid

### CODE REVIEW

**Comment by ronie** [comment]
looks good to me

File: xbmc/guilib/GUIBaseContainer.cpp [hunk 1]
```diff
@@ -1308,6 +1308,12 @@ std::string CGUIBaseContainer::GetLabel(int info) const
         label = StringUtils::Format("%u", numItems);
     }
     break;
+  case CONTAINER_ACTUAL_ITEMS:
```
Comment (BigNoid) [inline xbmc/guilib/GUIBaseContainer.cpp]: wouldn't it be easier to combine with ```CONTAINER_NUM_ITEMS```?

File: xbmc/GUIInfoManager.cpp [hunk 2]
```diff
@@ -2323,7 +2323,13 @@ const infomap mediacontainer[] = {{ "hasfiles",         CONTAINER_HASFILES },
...
 ///                  \anchor Container_NumItems
 ///                  _boolean_,
-///     Number of items in the container or grouplist with given id. If no id is
+///     Number of items in the container or grouplist with given id, excluding parent folder item. If no id is
+///     specified it grabs the current container.
+///   }
+///   \table_row3{   <b>`Container(id).ActualItems`</b>,
+///                  \anchor Container_ActualItems
+///                  _boolean_,
+///     Number of items in the container or grouplist with given id, including parent folder item. If no id is
```
Comment (AlwinEsch) [inline xbmc/GUIInfoManager.cpp]: Don't forget the `\` before the commas in text!

Is the bad part on doxygen if on a Macro (e.g. `\table_row3`) are more as one value, then the comma separate the values. If now a text is there also with them brings it problems.

File: xbmc/guilib/GUIBaseContainer.cpp (hunk 1 above)
Comment (phil65) [inline xbmc/guilib/GUIBaseContainer.cpp]: I dont think so, the way it is now only adds two lines of actual code (could even condense it into one).

File: xbmc/GUIInfoManager.cpp (hunk 2 above)
Comment (phil65) [inline xbmc/GUIInfoManager.cpp]: Thx, will fix!

File: xbmc/guilib/GUIBaseContainer.cpp (hunk 1 above)
Comment (BigNoid) [inline xbmc/guilib/GUIBaseContainer.cpp]: Combing is only one line extra:
```
    case CONTAINER_ACTUAL_ITEMS:
    case CONTAINER_NUM_ITEMS:
    {
      unsigned int numItems = GetNumItems();
      if (info == CONTAINER_NUM_ITEMS && numItems && m_items[0]->IsFileItem() && (std::static_pointer_cast<CFileItem>(m_items[0]))->IsParentFolder())
        label = StringUtils::Format("%u", numItems-1);
      else
        label = StringUtils::Format("%u", numItems);
    }
```

Comment (phil65) [inline xbmc/guilib/GUIBaseContainer.cpp]: Also fine for me, doesnt really matter here imo.

Comment (BigNoid) [inline xbmc/guilib/GUIBaseContainer.cpp]: I was told to avoid duplicate code in one of my pr's :)

Comment (phil65) [inline xbmc/guilib/GUIBaseContainer.cpp]: I know, but in those cases lot more logic was involved. ;)

Comment (phil65) [inline xbmc/guilib/GUIBaseContainer.cpp]: Anyways, updated to your suggestion.

**Comment by phil65** [comment]
jenkins build this please.

**Comment by phil65** [comment]
jenkins build and merge.

**Comment by ksooo** [comment]
@phil65: PLEASE, no! I guess you mean "current" items, not "actual"? "Actual" is a false friend for Germans as in German "aktuell" means "current". English "actual" means "tatsächlich"! 

Due to the large number of Germans in the project we have this language flaw at different places in the api and in the code, already. Please not one more of these, please.

**Comment by phil65** [comment]
that label was proposed by @HitcherUK who is native speaker.

**Comment by ksooo** [comment]
> that label was proposed by @HitcherUK who is native speaker.

 As I'm not a native English speaker... i will shut up, although it still looks strange to me.

**Comment by ksooo** [comment]
Seems I misunderstood the semantics of this new label. I was just explained what it actually is supposed for and yes, then the wording is correct. Sorry for the noise.

**Pull request merged**